OpenRA.Mods.Common/Activities/Parachute.cs

```diff
@@ -0,0 +1,99 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	public class Parachute : Activity
+	{
+		WPos dropPosition;
+		WPos currentPosition;
+		WVec fallVector;
+		bool triggered = false;
+		UpgradeManager um;
```

Comment:
You can make all of these cached traits `readonly`. (And iirc you'll have to move them to the top then to conform to the coding style. Run StyleCop to see if it complains about anything.)